cvmfs/repository.py

```diff
@@ -227,145 +107,64 @@ def verify(self, public_key_path):
         return History(history_db)
 
-    def switch_revision(self, revision):
+    def get_current_revision(self):
+        return self._get_revision_by_number(self.manifest.revision)
+
+    def get_revision(self, revision_data):
+        if isinstance(revision_data, int):
+            return self._get_revision_by_number(revision_data)
+        elif isinstance(revision_data, str):
+            return self._get_revision_by_tag(revision_data)
+
+    def _get_revision_by_number(self, revision):
         history = self.retrieve_history()
         revision_tag = history.get_tag_by_revision(revision)
-        self._root_catalog_hash = revision_tag.hash
+        return Revision(self, revision_tag)
 
-    def switch_tag(self, tag_name):
+    def _get_revision_by_tag(self, tag_name):
         history = self.retrieve_history()
         revision_tag = history.get_tag_by_name(tag_name)
-        self._root_catalog_hash = revision_tag.hash
+        return Revision(self, revision_tag)
 
     def retrieve_whitelist(self):
         """ retrieve and parse the .cvmfswhitelist file from the repository """
         whitelist = self._fetcher.retrieve_raw_file(_common._WHITELIST_NAME)
         return Whitelist(whitelist)
 
-
     def retrieve_certificate(self):
         """ retrieve the repository's certificate file """
         certificate = self.retrieve_object(self.manifest.certificate, 'X')
         return Certificate(certificate)
 
-
     def retrieve_object(self, object_hash, hash_suffix = ''):
         """ Retrieves an object from the content addressable storage """
         path = "data/" + object_hash[:2] + "/" + object_hash[2:] + hash_suffix
         return self._fetcher.retrieve_file(path)
 
-
-    def retrieve_root_catalog(self):
-        return self.retrieve_catalog(self._root_catalog_hash)
-
-
-    def retrieve_catalog_for_path(self, needle_path):
-        """ Recursively walk down the Catalogs and find the best fit for a path """
-        clg = self.retrieve_root_catalog()
-        while True:
-            new_nested_reference = clg.find_nested_for_path(needle_path)
-            if new_nested_reference is None:
-                break
-            nested_reference = new_nested_reference
-            clg = self.retrieve_catalog(nested_reference.hash)
-        return clg
-
-
     def close_catalog(self, catalog):
         try:
-            del self._opened_catalogs[catalog.hash]
+            del self.opened_catalogs[catalog.hash]
         except KeyError, e:
             print "not found:" , catalog.hash
-            pass
-
 
-    def retrieve_catalog(self, catalog_hash):
-        """ Download and open a catalog from the repository """
-        if catalog_hash in self._opened_catalogs:
-            return self._opened_catalogs[catalog_hash]
-        return self._retrieve_and_open_catalog(catalog_hash)
-
-    def _retrieve_and_open_catalog(self, catalog_hash):
+    def retrieve_and_open_catalog(self, catalog_hash):
```

Comment:
It makes sense to make this method name more specific. However, I would make it protected and add a `retrieve_catalog()` method to do the `opened_catalogs` lookup. The latter could be called by `Revision` for the sake of loose coupling and information hiding.